### PR TITLE
feat(api): pass assets to marxan run job

### PR DESCRIPTION
**Pull request by Dyostiq**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: api/apps/api/config/default.json [hunk 1]
```diff
@@ -13,6 +13,9 @@
      },
    "concurrency": 50
    },
+  "api": {
+    "url": "http://api:3000"
```
Comment (kgajowy) [inline api/apps/api/config/default.json:17]: @hotzevzl please keep in mind to set those on deployments.

File: api/apps/api/src/modules/scenarios/marxan-run/input-parameter-file.provider.ts [hunk 2]
```diff
@@ -78,13 +70,15 @@ class InputParameterFile {
...
+    @InjectRepository(Scenario)
+    private readonly scenarioRepository: Repository<Scenario>,
     @Inject(ioSettingsToken)
     private readonly ioSettings: IoSettings,
   ) {}
 
   async getInputParameterFile(scenarioId: string): Promise<string> {
-    const scenario = await this.scenariosService.getById(scenarioId);
+    const scenario = await this.scenarioRepository.findOne(scenarioId);
+    assertDefined(scenario);
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios/marxan-run/input-parameter-file.provider.ts:81]: shouldn't (isn't it already?) it be done on service layer?
`api/apps/api/src/modules/scenarios/scenarios.service.ts`

File: api/apps/api/src/modules/scenarios/marxan-run/assets.service.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { Assets } from '@marxan/scenario-run-queue';
+import { FactoryProvider, Inject, Injectable } from '@nestjs/common';
+import { AppConfig } from '@marxan-api/utils/config.utils';
+import { IoSettings, ioSettingsToken } from './io-settings';
+
+export const apiUrlToken = Symbol('api url token');
+export const apiUrlProvider: FactoryProvider<string> = {
+  provide: apiUrlToken,
+  useFactory: () => AppConfig.get<string>('api.url'),
+};
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios/marxan-run/assets.service.ts:10]: don't we usually keep those in standalone file? (like for api/apps/api/src/modules/scenarios/marxan-run/io-settings.ts below)

File: api/apps/api/src/modules/scenarios/marxan-run/run.service.ts [hunk 4]
```diff
@@ -44,14 +45,18 @@ export class RunService {
...
     private readonly scenarios: Repository<Scenario>,
+    private readonly assets: AssetsService,
   ) {
     queueEvents.on(`completed`, ({ jobId }) => this.handleFinished(jobId));
     queueEvents.on(`failed`, ({ jobId }) => this.handleFailed(jobId));
   }
 
   async run(scenarioId: string): Promise<void> {
+    const assets = await this.assets.forScenario(scenarioId);
+    assertDefined(assets);
```
Comment (kgajowy) [inline api/apps/api/src/modules/scenarios/marxan-run/run.service.ts:56]: As above, we shall keep it in top service which already does that (btw that assert wouldn't ever trigger, as AsserService does the check as well)

File: api/libs/scenario-run-queue/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+export const queueName = 'scenario-run-queue';
+export type Assets = {
+  url: string;
+  relativeDestination: string;
+}[];
```
Comment (kgajowy) [inline api/libs/scenario-run-queue/src/index.ts:5]: Would be great to add a subtask to use this in geoprocessing ❤️ 🙇

**Review by kgajowy — APPROVED** [review]

File: api/apps/api/src/modules/scenarios/marxan-run/assets.service.ts (hunk 3 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/scenarios/marxan-run/assets.service.ts:10]: io-settings previously also was in one file with its user. I tend to make it a separate files when it has more than one user.

File: api/apps/api/src/modules/scenarios/marxan-run/input-parameter-file.provider.ts (hunk 2 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/scenarios/marxan-run/input-parameter-file.provider.ts:81]: it is but I needed to narrow the type here, as `findOne` returns `| undefined`.

On the other hand, it's not in a transaction so it's possible that the assertion will actually throw :)

File: api/apps/api/src/modules/scenarios/marxan-run/run.service.ts (hunk 4 above)
Comment (Dyostiq) [reply api/apps/api/src/modules/scenarios/marxan-run/run.service.ts:56]: it's a leftover, thanks :)

File: api/apps/api/config/default.json (hunk 1 above)
Comment (hotzevzl) [reply api/apps/api/config/default.json:17]: yep, as long as they are in `README.md` and in `env.default` we're good - I think you added this one some time ago.

Comment (hotzevzl) [reply api/apps/api/config/default.json:17]: for values needed in CI, we also need to add them to `.env-test-e2e` (or GH secrets if applicable) and to pass them through via `docker-compose-test-e2e.ci.yml`

**Review by hotzevzl — APPROVED** [review]

**Pull request merged**